phpstan.neon

```diff
@@ -0,0 +1,15 @@
+includes:
+  - rules.neon
+  - ../../mglaman/phpstan-drupal/extension.neon
+
+parameters:
+	level: 1
+	paths:
+		- ../../../graphql
+	excludes_analyse:
+		- ../../../graphql/src/Entity/Server.php #Class GraphQL\Executor\ExecutionResult not found.
```

Comment:
you get this error because the onyx graphql vendor library might be missing locally?

I think you need the same setup as in travis CI: a drupal 8 checkout where this graphql module is in the modules folder + the composer dependencies installed.

Reply:
Hmm ich schaus mir nochmal an.

Reply:
Hab jetzt nochmal alles neu installiert, kriege immer noch diesen Error

Reply:
Weiters kommt das immer ganz unten als Error auf:

Child process error (exit code 255): PHP Fatal error:  During class fetch: Uncaught PHPStan\Broker\ClassAutoloadingException: Class Drupal\typed_data\DataFetcherTrait not found. in                 
     phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/src/Reflection/Runtime/RuntimeReflectionProvider.php:125                                                                          
     Stack trace:                                                                                                                                                                                         
     #0 [internal function]: PHPStan\Reflection\Runtime\RuntimeReflectionProvider->PHPStan\Reflection\Runtime\{closure}()                                                                                 
     #1 /home/dylan/Drupal8-Template/web/modules/graphql/src/GraphQL/Resolver/Path.php(18): spl_autoload_call()                                                                                           
     #2 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/vendor/composer/ClassLoader.php(444): include('/home/dylan/Dru...')                                                            
     #3 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/vendor/composer/ClassLoader.php(322): Composer\Autoload\includeFile()                                                          
     #4 [internal function]: Composer\Autoload\ClassLoader->loadClass()                                                                                                                                   
     #5 [internal function]: spl_autoload_call()                                                                                                                                                          
     #6 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/src/Reflection/Runtime/RuntimeReflectionProvider.php(129): class_exists()                                                      
     #7 p in /home/dylan/Drupal8-Template/web/modules/graphql/src/GraphQL/Resolver/Path.php on line 18                                                                                                    
     Fatal error: During class fetch: Uncaught PHPStan\Broker\ClassAutoloadingException: Class Drupal\typed_data\DataFetcherTrait not found. in                                                           
     phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/src/Reflection/Runtime/RuntimeReflectionProvider.php:125                                                                          
     Stack trace:                                                                                                                                                                                         
     #0 [internal function]: PHPStan\Reflection\Runtime\RuntimeReflectionProvider->PHPStan\Reflection\Runtime\{closure}()                                                                                 
     #1 /home/dylan/Drupal8-Template/web/modules/graphql/src/GraphQL/Resolver/Path.php(18): spl_autoload_call()                                                                                           
     #2 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/vendor/composer/ClassLoader.php(444): include('/home/dylan/Dru...')                                                            
     #3 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/vendor/composer/ClassLoader.php(322): Composer\Autoload\includeFile()                                                          
     #4 [internal function]: Composer\Autoload\ClassLoader->loadClass()                                                                                                                                   
     #5 [internal function]: spl_autoload_call()                                                                                                                                                          
     #6 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/src/Reflection/Runtime/RuntimeReflectionProvider.php(129): class_exists()                                                      
     #7 p in /home/dylan/Drupal8-Template/web/modules/graphql/src/GraphQL/Resolver/Path.php on line 18                                                                                                    
                                                                                                                                                                                                          
     Child process error (exit code 255): PHP Fatal error:  During class fetch: Uncaught PHPStan\Broker\ClassAutoloadingException: Class Drupal\typed_data\DataFetcherTrait not found. in                 
     phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/src/Reflection/Runtime/RuntimeReflectionProvider.php:125                                                                          
     Stack trace:                                                                                                                                                                                         
     #0 [internal function]: PHPStan\Reflection\Runtime\RuntimeReflectionProvider->PHPStan\Reflection\Runtime\{closure}()                                                                                 
     #1 /home/dylan/Drupal8-Template/web/modules/graphql/src/GraphQL/Resolver/Path.php(18): spl_autoload_call()                                                                                           
     #2 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/vendor/composer/ClassLoader.php(444): include('/home/dylan/Dru...')                                                            
     #3 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/vendor/composer/ClassLoader.php(322): Composer\Autoload\includeFile()                                                          
     #4 [internal function]: Composer\Autoload\ClassLoader->loadClass()                                                                                                                                   
     #5 [internal function]: spl_autoload_call()                                                                                                                                                          
     #6 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/src/Reflection/Runtime/RuntimeReflectionProvider.php(129): class_exists()                                                      
     #7 p in /home/dylan/Drupal8-Template/web/modules/graphql/src/GraphQL/Resolver/Path.php on line 18                                                                                                    
     Fatal error: During class fetch: Uncaught PHPStan\Broker\ClassAutoloadingException: Class Drupal\typed_data\DataFetcherTrait not found. in                                                           
     phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/src/Reflection/Runtime/RuntimeReflectionProvider.php:125                                                                          
     Stack trace:                                                                                                                                                                                         
     #0 [internal function]: PHPStan\Reflection\Runtime\RuntimeReflectionProvider->PHPStan\Reflection\Runtime\{closure}()                                                                                 
     #1 /home/dylan/Drupal8-Template/web/modules/graphql/src/GraphQL/Resolver/Path.php(18): spl_autoload_call()                                                                                           
     #2 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/vendor/composer/ClassLoader.php(444): include('/home/dylan/Dru...')                                                            
     #3 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/vendor/composer/ClassLoader.php(322): Composer\Autoload\includeFile()                                                          
     #4 [internal function]: Composer\Autoload\ClassLoader->loadClass()                                                                                                                                   
     #5 [internal function]: spl_autoload_call()                                                                                                                                                          
     #6 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/src/Reflection/Runtime/RuntimeReflectionProvider.php(129): class_exists()                                                      
     #7 p in /home/dylan/Drupal8-Template/web/modules/graphql/src/GraphQL/Resolver/Path.php on line 18                                                                                                    
                                                                                                                                                                                                          
     Child process error (exit code 255): PHP Fatal error:  During class fetch: Uncaught PHPStan\Broker\ClassAutoloadingException: Class Drupal\typed_data\DataFetcherTrait not found. in                 
     phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/src/Reflection/Runtime/RuntimeReflectionProvider.php:125                                                                          
     Stack trace:                                                                                                                                                                                         
     #0 [internal function]: PHPStan\Reflection\Runtime\RuntimeReflectionProvider->PHPStan\Reflection\Runtime\{closure}()                                                                                 
     #1 /home/dylan/Drupal8-Template/web/modules/graphql/src/Plugin/GraphQL/DataProducer/TypedData/PropertyPath.php(36): spl_autoload_call()                                                              
     #2 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/vendor/composer/ClassLoader.php(444): include('/home/dylan/Dru...')                                                            
     #3 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/vendor/composer/ClassLoader.php(322): Composer\Autoload\includeFile()                                                          
     #4 [internal function]: Composer\Autoload\ClassLoader->loadClass()                                                                                                                                   
     #5 [internal function]: spl_autoload_call()                                                                                                                                                          
     #6 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/src/Reflection/Runtime/RuntimeReflectionProvider. in                                                                           
     /home/dylan/Drupal8-Template/web/modules/graphql/src/Plugin/GraphQL/DataProducer/TypedData/PropertyPath.php on line 36                                                                               
     Fatal error: During class fetch: Uncaught PHPStan\Broker\ClassAutoloadingException: Class Drupal\typed_data\DataFetcherTrait not found. in                                                           
     phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/src/Reflection/Runtime/RuntimeReflectionProvider.php:125                                                                          
     Stack trace:                                                                                                                                                                                         
     #0 [internal function]: PHPStan\Reflection\Runtime\RuntimeReflectionProvider->PHPStan\Reflection\Runtime\{closure}()                                                                                 
     #1 /home/dylan/Drupal8-Template/web/modules/graphql/src/Plugin/GraphQL/DataProducer/TypedData/PropertyPath.php(36): spl_autoload_call()                                                              
     #2 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/vendor/composer/ClassLoader.php(444): include('/home/dylan/Dru...')                                                            
     #3 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/vendor/composer/ClassLoader.php(322): Composer\Autoload\includeFile()                                                          
     #4 [internal function]: Composer\Autoload\ClassLoader->loadClass()                                                                                                                                   
     #5 [internal function]: spl_autoload_call()                                                                                                                                                          
     #6 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/src/Reflection/Runtime/RuntimeReflectionProvider. in                                                                           
     /home/dylan/Drupal8-Template/web/modules/graphql/src/Plugin/GraphQL/DataProducer/TypedData/PropertyPath.php on line 36                                                                               
                                                                                                                                                                                                          
     Child process error (exit code 255): PHP Fatal error:  Declaration of Drupal\KernelTests\KernelTestBase::setUpBeforeClass() must be compatible with PHPUnit\Framework\TestCase::setUpBeforeClass():  
     void in /home/dylan/Drupal8-Template/web/core/tests/Drupal/KernelTests/KernelTestBase.php on line 227                                                                                                
     Fatal error: Declaration of Drupal\KernelTests\KernelTestBase::setUpBeforeClass() must be compatible with PHPUnit\Framework\TestCase::setUpBeforeClass(): void in                                    
     /home/dylan/Drupal8-Template/web/core/tests/Drupal/KernelTests/KernelTestBase.php on line 227                                                                                                        
                                                                                                                                                                                                          
     Child process error (exit code 255): PHP Fatal error:  During class fetch: Uncaught PHPStan\Broker\ClassAutoloadingException: Class Drupal\typed_data\DataFetcherTrait not found. in                 
     phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/src/Reflection/Runtime/RuntimeReflectionProvider.php:125                                                                          
     Stack trace:                                                                                                                                                                                         
     #0 [internal function]: PHPStan\Reflection\Runtime\RuntimeReflectionProvider->PHPStan\Reflection\Runtime\{closure}()                                                                                 
     #1 /home/dylan/Drupal8-Template/web/modules/graphql/src/GraphQL/ResolverBuilder.php(22): spl_autoload_call()                                                                                         
     #2 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/vendor/composer/ClassLoader.php(444): include('/home/dylan/Dru...')                                                            
     #3 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/vendor/composer/ClassLoader.php(322): Composer\Autoload\includeFile()                                                          
     #4 [internal function]: Composer\Autoload\ClassLoader->loadClass()                                                                                                                                   
     #5 [internal function]: spl_autoload_call()                                                                                                                                                          
     #6 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/src/Reflection/Runtime/RuntimeReflectionProvider.php(129): class_exists()                                                      
     #7 in /home/dylan/Drupal8-Template/web/modules/graphql/src/GraphQL/ResolverBuilder.php on line 22                                                                                                    
     Fatal error: During class fetch: Uncaught PHPStan\Broker\ClassAutoloadingException: Class Drupal\typed_data\DataFetcherTrait not found. in                                                           
     phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/src/Reflection/Runtime/RuntimeReflectionProvider.php:125                                                                          
     Stack trace:                                                                                                                                                                                         
     #0 [internal function]: PHPStan\Reflection\Runtime\RuntimeReflectionProvider->PHPStan\Reflection\Runtime\{closure}()                                                                                 
     #1 /home/dylan/Drupal8-Template/web/modules/graphql/src/GraphQL/ResolverBuilder.php(22): spl_autoload_call()                                                                                         
     #2 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/vendor/composer/ClassLoader.php(444): include('/home/dylan/Dru...')                                                            
     #3 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/vendor/composer/ClassLoader.php(322): Composer\Autoload\includeFile()                                                          
     #4 [internal function]: Composer\Autoload\ClassLoader->loadClass()                                                                                                                                   
     #5 [internal function]: spl_autoload_call()                                                                                                                                                          
     #6 phar:///home/dylan/Drupal8-Template/vendor/phpstan/phpstan/phpstan/src/Reflection/Runtime/RuntimeReflectionProvider.php(129): class_exists()                                                      
     #7 in /home/dylan/Drupal8-Template/web/modules/graphql/src/GraphQL/ResolverBuilder.php on line 22

Reply:
you need to install the dependencies of graphql module, see https://github.com/drupal-graphql/graphql/blob/8.x-4.x/.travis.yml#L90
```
composer require webonyx/graphql-php:^0.13.1 drupal/typed_data:^1.0
```